oura-library/src/main/kotlin/org/radarbase/oura/converter/PayloadToSourceRecordConverter.kt

```diff
@@ -0,0 +1,24 @@
+package org.radarbase.oura.converter
+
+import okhttp3.Headers
+import org.radarbase.oura.request.RestRequest
+import java.io.IOException
+import java.time.Duration
+import java.time.Instant
+
+interface PayloadToSourceRecordConverter {
+    @Throws(IOException::class)
+    fun convert(
+            request: RestRequest, headers: Headers, data: ByteArray
+    ): Sequence<Result<TopicData>>
```

Comment:
Replace with `Sequence<Pair<IndexedRecord,IndexedRecord>>`, where the first Indexed Record is the key and the second is the value.